### ISSUE TITLE
Types wrongly redefined in early 0.5.0-dev

### ISSUE THREAD

**Issue by damiendr** [JuliaArchive/LegacyStrings.jl#6]
I am stuck at a particular version of Julia (`0.5.0-dev+2349`) where UTF-16 and UTF-32 types are still defined in `Base` (that's pre PR [#16590](https://github.com/JuliaLang/julia/pull/16590)). This creates problems because `LegacyString` then redefines these types.

I used the following fix in `LegacyStrings.jl`, line 45:
```
+    if !isdefined(Base, :UTF16String)
-    if VERSION >= v"0.5.0-"
```

I guess this is a fairly rare case but it might be worth relying more on `isdefined` than version checks?

**Comment by tkelman** [comment]
Yes. Submit a PR? There are also some annoying "cannot import `lastidx`" warnings I've seen due to this package on some Julia versions.

**Comment by damiendr** [comment]
OK, I'll submit a PR for the issue with redefined types! I don't know about the "cannot import `lastidx`" problem though.

**Comment by tkelman** [comment]
No worries, anything where you know a good solution, a PR is much appreciated. I'll try to remember to dig into where the lastidx thing comes from next time I see it happen.

**Comment by stevengj** [comment]
No longer relevant.
